src/chapters/more-on-functions/recursion-walkthrough.rst

```diff
@@ -0,0 +1,207 @@
+Recursion Walkthrough: The Base Case
+=====================================
+
+To ease into the concept of recursion, let's start with a loop task.
+
+In the Arrays chapter, we examined  the ``join`` method, which combines the
+elements of an array into a single string. If we have
+``arr = ['L', 'C', '1', '0', '1']``, then ``arr.join('')`` returns the string
+``'LC101'``.
+
+We can reproduce this action with either a ``for`` or a ``while`` loop.
+
+Joining Array Elements With a Loop
+-----------------------------------
+
+Examine the code samples below:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Use a ``for`` loop to iterate through the array and add each entry into the ``newString`` variable.
+     - Use a ``while`` loop to add the first element in the array to ``newString``, then remove that element from the array.
+
+   * - .. sourcecode:: js
+
+         let arr = ['L', 'C', '1', '0', '1'];
+         let newString = '';
+
+         for (i = 0; i < arr.length; i++){
+            newString = newString + arr[i];
+         }
+
+         console.log(newString);
+         console.log(arr);
+
+       **Output**
+
+       .. sourcecode:: js
+
+         'LC101'
+         ['L', 'C', '1', '0', '1']
+
+     - .. sourcecode:: js
+
+         let arr = ['L', 'C', '1', '0', '1'];
+         let newString = '';
+
+         while (arr.length > 0){
+            newString += arr[0];
+            arr.shift();
+         }
+         console.log(newString);
+         console.log(arr);
+
+       **Output**
+
+       .. sourcecode:: js
+
+         'LC101'
+         [ ]
+
+Inside each loop, the code simply adds two strings together - whatever is
+stored in ``newString`` plus one element from the array. In the ``for`` loop,
+the element is the next item in the sequence of entries.  In the ``while``
+loop, the element is always the first entry from whatever remains in the array.
+
+OK, the loops join the array elements together. Now let's see how to
+accomplish the same task without a ``for`` or ``while`` statement.
+
+Bring In Recursion Concepts
+----------------------------
+
+First, state the problem to solve: *Combine the elements from an array into a
+string*.
+
+Second, split the problem into small, identical steps: Looking at the loops
+above, the "identical step" is just adding two strings together - ``newString``
+and the next entry in the array.
+
+Third, build a function to accomplish the small steps: Let's call the function
+``combineEntries``, and we will set an array as the parameter.
+
+.. sourcecode:: js
+
+   function combineEntries(arrayName){
+      //some code here
```

Comment:
Consider making this `// TODO` to indicate that we'll be filling this in as we go. In other places in the book, we simply omit a function body with a comment like the one you're using here, indicating that there really is code there, but we just aren't showing it. That isn't the case here.

Reply:
Done.